### ISSUE TITLE
Passing in user-specified test names

### ISSUE THREAD

**Issue by katef** [silentbicycle/greatest#68]
Hi,

I have some tests which share the same function, and that function is passed the item to test as its `void *ENV` argument. I added this, so I can pass a useful name for each of those tests:
```
+/* adapted from GREATEST_RUN_TEST1() */
+#define GREATEST_RUN_TEST1_NAME(NAME, TEST, ENV) \
+    do {                                                                \
+        if (greatest_test_pre((NAME)) == 1) {                            \
+            enum greatest_test_res res = GREATEST_SAVE_CONTEXT();       \
+            if (res == GREATEST_TEST_RES_PASS) {                        \
+                res = TEST((ENV)) ;                        \
+            }                                                           \
+            greatest_test_post((NAME), res);                             \
+        }                                                               \
+    } while (0)
```
where `NAME` replaces `#TEST`, which was the same for each test.

I don't like making the API bigger just for this, because greatest's appeal is its balance of size and usefulness.

Example of the kind of array I have: https://github.com/katef/jvst/blob/master/tests/unit/test_ir.c#L137
I'll probably make them independent functions, and live with the duplication of setup and teardown inside each. I wanted to share the situation I have, in case it's interesting.

**Comment by silentbicycle** [comment]
I don't really follow the example you linked to, in isolation, but there may be other ways to work with existing API -- `CHECK_CALL` may help.

(At first, this sounds related to #63, but I don't think it is -- that's about being able to pass along a suffix to parametric tests, among other things.)

**Comment by silentbicycle** [comment]
Based on direct communication, it sounds like this should be addressed by the same changes as #63. Closing.
